src/main/java/io/prestosql/plugin/tiledb/TileDBPageSink.java

```diff
@@ -415,52 +420,55 @@ else if (DATE.equals(type)) {
             long value;
             switch (colType) {
                 case TILEDB_DATETIME_AS: {
-                    value = new Timestamp(0).toInstant().atOffset(ZoneOffset.UTC).plusNanos((long) ((type.getLong(block, position)) * 0.0001)).toInstant().toEpochMilli();
+                    value = type.getLong(block, position) * 10000000;
```

Comment:
I believe these conversions are not right. Every type is a 10^3 different, but you only have a `*10` difference between AS and FS. Please double check all conversions